server/src/main/java/org/opensearch/action/search/TransportDeletePITAction.java

```diff
@@ -0,0 +1,172 @@
+
+    @Inject
+    public TransportDeletePITAction(
+        SearchService searchService,
+        TransportService transportService,
+        ActionFilters actionFilters,
+        NamedWriteableRegistry namedWriteableRegistry,
+        TransportSearchAction transportSearchAction,
+        ClusterService clusterService,
+        SearchTransportService searchTransportService
+    ) {
+        super(DeletePITAction.NAME, transportService, actionFilters, DeletePITRequest::new);
+        this.searchService = searchService;
+        this.namedWriteableRegistry = namedWriteableRegistry;
+        this.transportSearchAction = transportSearchAction;
+        this.clusterService = clusterService;
+        this.searchTransportService = searchTransportService;
+    }
+
+    @Override
+    protected void doExecute(Task task, DeletePITRequest request, ActionListener<DeletePITResponse> listener) {
+        List<SearchContextIdForNode> contexts = new ArrayList<>();
+        List<String> pitIds = request.getPitIds();
+        if (pitIds.size() == 1 && "_all".equals(pitIds.get(0))) {
+            deleteAllPits(listener);
+        } else {
+            for (String pitId : request.getPitIds()) {
+                SearchContextId contextId = SearchContextId.decode(namedWriteableRegistry, pitId);
+                contexts.addAll(contextId.shards().values());
+            }
+            deletePits(contexts, ActionListener.wrap(r -> {
+                if (r == contexts.size()) {
+                    listener.onResponse(new DeletePITResponse(true));
+                } else {
+                    logger.debug(
+                        () -> new ParameterizedMessage("Delete PITs failed. " + "Cleared {} contexts out of {}", r, contexts.size())
+                    );
+                    listener.onResponse(new DeletePITResponse(false));
+                }
+            }, e -> {
+                logger.debug("Delete PITs failed ", e);
+                listener.onResponse(new DeletePITResponse(false));
+            }));
+        }
+    }
+
+    /**
+     * Delete all active PIT reader contexts
+     */
+    void deleteAllPits(ActionListener<DeletePITResponse> listener) {
+        int size = clusterService.state().getNodes().getSize();
+        ActionListener groupedActionListener = getGroupedListener(listener, size);
+        for (final DiscoveryNode node : clusterService.state().getNodes()) {
+            try {
+                Transport.Connection connection = searchTransportService.getConnection(null, node);
+                searchTransportService.sendDeleteAllPitContexts(connection, groupedActionListener);
+            } catch (Exception e) {
+                groupedActionListener.onFailure(e);
+            }
+        }
+    }
+
+    /**
+     * Delete list of pits, return success if all reader contexts are deleted ( or not found ).
+     */
+    void deletePits(List<SearchContextIdForNode> contexts, ActionListener<Integer> listener) {
```

Comment:
can we move this to a separate handler class which would enable us to do better functional testing?

Reply:
unlike create pit (where mocking the transport action didn't give us the desired results), i was able to mock and write functional tests without segregating the methods into a separate file.